Pokemon/Pokemon/PokemonTableViewController.m

```diff
@@ -0,0 +1,506 @@
+                          @"Haunter",
+                          @"Gengar",
+                          @"Koffing",
+                          @"Weezing",
+                          @"Exeggcute",
+                          @"Exeggutor",
+                          @"Tangela",
+                          @"Paras",
+                          @"Parasect",
+                          @"Dratini",
+                          @"Dragonair",
+                          @"Dragonite",
+                          @"Pidgey",
+                          @"Pidgeotto",
+                          @"Pidgeot",
+                          @"Rattata",
+                          @"Raticate",
+                          @"Spearow",
+                          @"Fearow",
+                          @"Jigglypuff",
+                          @"Wigglytuff",
+                          @"Meowth",
+                          @"Persian",
+                          @"Farfetch",
+                          @"Doduo",
+                          @"Dodrio",
+                          @"Lickitung",
+                          @"Chansey",
+                          @"Kangaskhan",
+                          @"Tauros",
+                          @"Ditto",
+                          @"Eevee",
+                          @"Porygon",
+                          @"Snorlax",
+                          @"Geodude",
+                          @"Graveler",
+                          @"Golem",
+                          @"Onix",
+                          @"Rhyhorn",
+                          @"Rhydon",
+                          @"Aerodactyl",
+                          @"Clefairy",
+                          @"Clefable",
+                          @"Mankey",
+                          @"Primeape",
+                          @"Machop",
+                          @"Machoke",
+                          @"Machamp",
+                          @"Hitmonlee",
+                          @"Hitmonchan"];
+
+}
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+
+#pragma mark - Table view data source
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+#warning Potentially incomplete method implementation.
+    
+    NSArray *keys = [self.pokemon allKeys];
+    
+    return keys.count;
+    
+    
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+#warning Incomplete method implementation.
+    
+    NSArray *keys = [self.pokemon allKeys];
+    NSString *words = keys[section];
+    NSArray *words2 = [self.pokemon objectForKey:words];
+    
+    return words2.count;
+    
+    
+}
+
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"pokemonCell" forIndexPath:indexPath];
+    
+    NSArray *keys = [self.pokemon allKeys];
```

Comment:
This logic occurs a couple of times in this file; you might want to make a separate method to avoid the duplication.